### PR TITLE
[wip] Add About dialog

### PR DESCRIPTION
**Pull request by ralphtheninja**
Closes https://github.com/deltachat/deltachat-desktop/issues/103

Currently looks really crappy since long values doesn't break properly and are rendered outside the dialog. But at least we have the complete flow and data ready.

screenshot from 2018-10-21 04-27-26

What's left:

- [ ] fix layout (css, or replace table with something better, suggestions welcome, basically `hilfe!`)
- [ ] the user _must_ be logged in to get the info so the menu alternative must be disabled when logged out
- [ ] `the_underscore_keys` should be translated

cc @karissa @Jikstra @hpk42

### CODE REVIEW

File: src/main/ipc.js [hunk 1]
```diff
@@ -58,6 +58,10 @@ function init (cwd) {
     e.returnValue = ret
   })
 
+  ipc.on('dispatchSyncNoRender', (e, ...args) => {
+    e.returnValue = dispatch(...args)
+  })
```
Comment (ralphtheninja) [inline src/main/ipc.js]: Needed a separate message to be able to get data from the backend _without_ rendering automatically, i.e. get data outside of the normal state pipeline.

**Comment by Jikstra** [comment]
> the user must be logged in to get the info so the menu alternative must be disabled when logged out

Hmm, why not only show the keys which we can display if the user is logged in? Not having an about when not logged in seems a bit weird to me. What's blocking us from always showing it?

**Comment by ralphtheninja** [comment]
The current design doesn't allow us to query the DC context because it's created in the login method. We must tweak that first.

@hpk requested this, hence why I put it in about dialog. I both agree and don't agree. Might as well have an advanced tab in the about dialog, many programs does this. Why do we have to follow exactly what signal does?

At the same time I don't really care either, as long as we don't have to bikeshed everything.

**Comment by ralphtheninja** [comment]
I'll leave this hanging for a while if anyone else has comments before closing.

**Comment by Jikstra** [comment]
> The current design doesn't allow us to query the DC context because it's created in the login method. We must tweak that first.

Makes sense :)

> Might as well have an advanced tab in the about dialog, many programs does this. 

Sounds like a good solution. Most people won't care about those infos, might be important to have a better understanding of the user environment on bug reports.

**Comment by ralphtheninja** [comment]
Hmm, multiple accounts make querying the dc context a bit problematic. Since we need to know the account, to get the folder in order to query it. Gah.

**Comment by Jikstra** [comment]
Why did you close/delete the branch?

**Pull request closed without merging**